app/src/main/java/com/sdp13epfl2021/projmag/video/SubtitleBuilder.kt

```diff
@@ -26,32 +29,53 @@ class SubtitleBuilder {
         }
     }
 
+    /**
+     * The start of the current subtitle
+     */
     var start: Int = 0
-        private set
+        set(pos) {
+            field = pos
+            if (end < pos) {
+                end = pos
+            }
+        }
+
+    /**
+     * the end of the current subtitle
+     */
     var end: Int = 0
-        private set
+        set(pos) {
+            if (pos < start) {
+                start = pos
+            }
+            field = pos
+        }
 
     private val subtitle: MutableList<SubtitlePart> = mutableListOf()
 
-    fun setStartOrEnd(or: Boolean, pos: Int) {
-        if (or == START) {
-            start = pos
-            end = if (end < pos) pos else end
-        } else {
-            start = if (pos < start) pos else start
-            end = pos
-        }
-    }
-
+    /**
+     * Add text to the current subtitle with `start` and `end` timestamps
+     *
+     * @param content the text of the subtitle
+     */
     fun add(content: String) {
         subtitle.add(SubtitlePart(content, start, end))
     }
 
+    /**
+     * Build the subtitle file and returns it
+     *
+     * @return the subtitle file
+     */
     fun build(): String {
         val str = "$WEBVTT\n\n"
         return subtitle.fold(str) { s, e -> s + (e.toString()) }
     }
 
+    fun setStartOrEnd(start: Any, defaultTimestamp: Int) {
+
+    }
```

Comment:
Is it normal that this function is empty ?

Reply:
It has been safe deleted, in this [commit](https://github.com/asfoury/projmag/pull/190/commits/507fa18a47c00e5f1ab44695b47cae1ce7ab1537)

Also thanks a lot for finding the issue with the tests :+1: